File: ExampleJsonPost-py/ExampleJsonPost.py
import sys
import os
import requests
import json

async def main(args):
    base_url = "https://optimizer2.routesavvy.com/RSAPI.svc/"

    try:
        if len(args) != 3:
            print("Usage: <input json request file> <output json result file>")
            input("Press Enter to exit.")
            return

        in_file_path = args[1]
        out_file_path = args[2]

        with open(in_file_path, 'r') as file:
            request_str = file.read().replace("\t", "")

        endpoint = "POSTOptimize"
        request_url = f"{base_url}{endpoint}"

        headers = {"Content-Type": "application/json"}
        response = requests.post(request_url, data=request_str, headers=headers)

        if response.status_code == 200:
            response_body = response.json()
            with open(out_file_path, 'w') as file:
                json.dump(response_body, file, indent=4)
            print("Success")
        else:
            print(f"Error: {response.status_code} - {response.reason}")

    except Exception as e:
        print(f"Exception: {str(e)}")

    input("Press Enter to exit.")

if __name__ == "__main__":
    asyncio.run(main(sys.argv))

File: ExampleJsonPost-php/ExampleJsonPost.php
<?php

$baseUrl = "https://optimizer2.routesavvy.com/RSAPI.svc/";
$POST = true;

try {
    if (count($argv) != 3) {
        echo "Usage: php script.php <input json request file> <output json result file>\n";
        echo "Press any key to exit.\n";
        readline();
        exit;
    }

    $inFilePath = $argv[1];
    $outFilePath = $argv[2];

    $requestStr = file_get_contents($inFilePath);
    $requestStr = str_replace("\t", "", $requestStr);

    $endpoint = "POSTOptimize";
    $requestUrl = $baseUrl . $endpoint;

    $httpClient = new \GuzzleHttp\Client();

    if ($POST) {
        $response = $httpClient->post($requestUrl, [
            'headers' => ['Content-Type' => 'application/json'],
            'body' => $requestStr
        ]);
    } else {
        // For GET request
        $requestUrl = $baseUrl . "GETOptimize?query=" . $requestStr;
        $response = $httpClient->get($requestUrl);
    }

    $statusCode = $response->getStatusCode();
    $responseBody = $response->getBody()->getContents();

    if ($statusCode == 200) {
        $responseObject = json_decode($responseBody);

        // Access properties of the response object
        $message = $responseObject->Message;

        if ($message !== null && $message !== "Success") {
            echo "An error occurred: $message\n";
        } else {
            try {
                $outputJson = json_decode($responseBody, true);
                file_put_contents($outFilePath, json_encode($outputJson, JSON_PRETTY_PRINT));
                echo "Success\n";
            } catch (Exception $ex) {
                echo "An error occurred: " . $ex->getMessage() . "\n";
            }
        }
    } else {
        echo "Error: $statusCode - " . $response->getReasonPhrase() . "\n";
    }
} catch (Exception $e) {
    echo "Exception: " . $e->getMessage() . "\n";
}

echo "Press any key to exit.\n";
readline();
